crates/typst/src/export/pdf/mod.rs

```diff
@@ -260,6 +260,7 @@ fn write_page_labels(ctx: &mut PdfContext) -> Vec<(NonZeroUsize, Ref)> {
 
 /// Compress data with the DEFLATE algorithm.
 #[tracing::instrument(skip_all)]
+#[comemo::memoize]
 fn deflate(data: &[u8]) -> Vec<u8> {
```

Comment:
This does a lot of hashing and cloning. The hashing we can't really prevent, but the cloning we can with `Arc<Vec<u8>>`. 

I also wonder how this affects memory usage as we're also compressing fonts here, which are already memoized separately. I think it should be okay if it returns `Arc`. Then, `subset_font`'s return type should be changed from `Bytes` to `Arc` though as the conversion is not free. (And the `Bytes` usage there is unnecessary anyway as it incurs prehashing.)